doc: Enhance rule documentation (#50)
salesforce/eslint-plugin-lwc: docs/rules/no-document-location.md

# Disallow document.location (no-document-location)

> The value of `document.location` in Lightning Locker vNext is `null`. To prevent code from breaking `window.location` should be used.

Reviewer: ```suggestion
The value of `document.location` in Lightning Locker vNext is `null`. To prevent code from breaking, use `window.location`.
```

> The following `document.location` references are prohibited or fixable to `window.location`:

Reviewer: ```suggestion
Replace these references with `window.location`:
```
